lib/matplotlib/axes/_base.py

```diff
@@ -685,20 +684,21 @@ def __getstate__(self):
         # that point.
         state = super().__getstate__()
         # Prune the sharing & twinning info to only contain the current group.
-        for grouper_name in [
-                '_shared_x_axes', '_shared_y_axes', '_twinned_axes']:
-            grouper = getattr(self, grouper_name)
-            state[grouper_name] = (grouper.get_siblings(self)
-                                   if self in grouper else None)
+        state["_shared_axes"] = {
```

Comment:
Does this mean we break pickling-compatibility between old and implementation? If so this needs an API-change note.

Reply:
The pickling representation is explicitly not stable (see `__mpl_version__` in `Figure.__setstate__`), changes fairly regularly without API note (nearly every time we add an attribute to Figure or Axes, we make old pickles unloadable on new versions, as they would be missing that attribute), and I wouldn't want to start pretending that we put *any* care in making the pickles stable.